okio/src/commonMain/kotlin/okio/-CommonPlatform.kt

```diff
@@ -23,7 +23,11 @@ internal expect fun String.asUtf8ToByteArray(): ByteArray
 // TODO make internal https://youtrack.jetbrains.com/issue/KT-37316
 expect class ArrayIndexOutOfBoundsException(message: String?) : IndexOutOfBoundsException
 
-internal expect inline fun <R> synchronized(lock: Any, block: () -> R): R
+expect class ALock
+
+internal expect fun newLock(): ALock
+
+internal expect inline fun <R> synchronized(lock: ALock, block: () -> R): R
```

Comment:
I do not like this syntactically, as it looks like the built-in function. I’d rather we use a new name, perhaps dereferencing ALock: `lock.withLock { ... }`

Reply:
(What we had before was attempting to maintain Java `synchronized` behavior; now we’re not)